### PR TITLE
 Add IDesktopWallpaper interface to shobjidl_core. 

### PR DESCRIPTION
**Pull request by AlphaModder**
I needed this interface for a project, and saw that so does @MOZGIII in #722, so I decided to write up the bindings myself. 

Pretty sure I've got everything right, but there is one thing: in the SDK, `IShellItemArray` is defined in `shobjidl_core.h`, but in `winapi` it's under `shobjidl.rs`. Because `IDesktopWallpaper` uses it, currently this PR introduces a dependency on the `shobjidl` feature into the `shobjidl_core` features, which may not be ideal.

Since both leaving `IShellItemArray` where it is and introducing a dependency or moving it to `shobjidl_core.rs` (restoring consistency with the SDK) would result in a breaking change, I suggest that if backwards compatibility is a concern, we move it to `shobjidl_core.rs`, but reexport it in `shobjidl.rs`.

### CODE REVIEW

**Comment by retep998** [comment]
>I suggest that if backwards compatibility is a concern, we move it to shobjidl_core.rs, but reexport it in shobjidl.rs.

This is the correct course of action. If a Windows SDK update caused things to be moved, then they should be defined in their new location and re-exported in their old location to prevent breaking changes (until a new major version of `winapi` is published).

**Comment by AlphaModder** [comment]
@retep998 Wow, thanks for the quick reply. I'll set up the reexport immediately.

**Comment by AlphaModder** [comment]
@retep998 Hmm, I've added the reexport, but it puts some new imports in `shobjidl_core` (`propsys` and `propkeydef`), which go unresolved. Do I need to specify a dependency somewhere?

**Comment by retep998** [comment]
You have to update the feature array in `build.rs`.

**Comment by AlphaModder** [comment]
@retep998 Got it! Thanks.

File: src/um/shobjidl.rs [hunk 1]
```diff
@@ -3,60 +3,16 @@
...
-    ) -> HRESULT,
-    fn GetItemAt(
-        dwIndex: DWORD,
-        ppsi: *mut *mut IShellItem,
-    ) -> HRESULT,
-    // TODO: Add IEnumShellItems
-    //fn EnumItems(
-    //    ppenumShellItems: *mut *mut IEnumShellItems,
-    //) -> HRESULT,
-}}
```
Comment (retep998) [inline src/um/shobjidl.rs:59]: Don't forget to actually reexport the stuff that was moved!

Comment (AlphaModder) [reply src/um/shobjidl.rs:59]: Ah, I did it in the `pub use` above. Would a `pub type` be better?

Comment (retep998) [reply src/um/shobjidl.rs:59]: No, `pub use` is fine. It should go in the spot where the definitions used to exist though, with a `// FIXME` stating to remove them in the next major version.

Comment (AlphaModder) [reply src/um/shobjidl.rs:59]: Ah, got it.

**Comment by MOZGIII** [comment]
This is great, looking forward to use it!

**Comment by AlphaModder** [comment]
@retep998 Is there anything left to do here, or is this ready to merge?

**Comment by MOZGIII** [comment]
I've implemented my feature using @AlphaModder's branch, and it works for me.
You can take a look here: https://github.com/MOZGIII/wallpaper-rs/tree/master/wallpaper-windows-shobjidl
At this point, whe only thing preventing me from publishing my crate is the dependency on the git branch instead of the proper crates.io-pubished version.
Can we merge this?

**Review by retep998 — APPROVED** [review]

**Pull request merged**